pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -131,6 +152,9 @@ func (dswp *desiredStateOfWorldPopulator) Run(sourcesReady config.SourcesReady,
 	dswp.hasAddedPodsLock.Lock()
 	dswp.hasAddedPods = true
 	dswp.hasAddedPodsLock.Unlock()
+
+	go dswp.cleanupFailedProcessPodMessage(stopCh)
```

Comment:
I haven't thought deeply about this, but I wonder if there is a more reliable way we can cleanup besides polling periodically.

Can we detect when the pod is removed from kubelet's list and clean up at that time?

Reply:
Kubelet's PodManager  does not provide any events, only Get methods. I was not able to quickly find the place where a new container is started when a pod is scheduled though.